onnxruntime/core/providers/openvino/backends/basic_backend.h

```diff
@@ -30,6 +30,16 @@ struct ov_tensor_data_t {
   const void* ort_ptr;
 };
 
+struct DynamicFlags {
```

Comment:
This structure should hold all the data members & methods related to IO shapes basically classify & operating on them for both static & dynamic shape

```suggestion
struct IOShapeManager {
   bool is_static = true;
   bool has_fully_dynamic = false;
   bool has_bounded_dynamic = false;  
   bool is_dynamic = has_fully_dynamic  && has_bounded_dynamic;
   
   void ValidateOrtDimsAgainstPartialShape(const std::vector<int64_t>& ort_dims,
                                         const ov::PartialShape& partial_shape) const;
     reshape_t OpenVINOParserUtils::ParseInputShape(const std::string& reshape_input_definition);
          };                              
```